pkg/scheduler/preemption/preemption.go

```diff
@@ -74,7 +75,7 @@ func candidatesOnlyFromQueue(candidates []*workload.Info, clusterQueue string) [
 }
 
 // GetTargets returns the list of workloads that should be evicted in order to make room for wl.
-func (p *Preemptor) GetTargets(wl workload.Info, assignment flavorassigner.Assignment, snapshot *cache.Snapshot) []*workload.Info {
+func (p *Preemptor) GetTargets(wl workload.Info, assignment flavorassigner.Assignment, snapshot *cache.Snapshot, borrowWithinCohort *kueue.BorrowWithinCohort) []*workload.Info {
 	resPerFlv := resourcesRequiringPreemption(assignment)
 	cq := snapshot.ClusterQueues[wl.ClusterQueue]
 
```

Comment:
I don't think so, suppose you have preemption config:
```golang
	  ReclaimWithinCohort: kueue.PreemptionPolicyLowerPriority,
	  BorrowWithinCohort: &kueue.BorrowWithinCohort{
		  Policy: kueue.BorrowWithinCohortPolicyLowerPriority,
		  MaxPriorityThreshold: MaxPriorityThreshold: ptr.To(100),
	  }
```
The new workload (w1) is priority 300, and we have an already admitted workload (w2) with priority 200. We need to return w2 from `findCandidates`, because its preemption may not require us to borrow, and its priority is below the priority of the new workload.